src/drive/web/modules/navigation/duck/actions.jsx

```diff
@@ -545,14 +545,22 @@ export const openFileWith = (id, filename) => {
   }
 }
 
-export const decryptVaultEncryptionKey = passphrase => {
+export const decryptVaultEncryptionKey = (vault, passphrase) => {
   return async (dispatch, _, { client }) => {
-    // TODO derive secret key + get encrypted vault key + decrypt it
     const salt = client.getStackClient().uri
-    const key = await deriveKey(passphrase, salt)
+    const derivedKey = await deriveKey(passphrase, salt)
+    const wrappedKey = decodeArrayBuffer(vault.key.encrypted_key)
+    const vaultKey = await unwrapAESKey(wrappedKey, derivedKey, {
+      algorithm: 'AES-GCM',
```

Comment:
The vaultKey is an AES-KW key normally : why having set AES-GCM ? If it's because of the potential bug in webcrypto not allowing an unwrapping to a KW, then I suggest using the "export / import" trick